### ISSUE TITLE
wsgi errors

### ISSUE THREAD

**Issue by rlucassen2** [Kozea/Radicale#562]
I installed Radicale from git on a Debian Stretch system, but I get these errors in the debug logs:

     'SERVER_SOFTWARE': 'WSGIServer/0.2',
     'wsgi.errors': <_io.StringIO object at 0xb61b8120>,
     'wsgi.file_wrapper': <class 'wsgiref.util.FileWrapper'>,
     'wsgi.input': <_io.BufferedReader name=7>,
     'wsgi.multiprocess': False,
     'wsgi.multithread': True,
     'wsgi.run_once': False,
     'wsgi.url_scheme': 'http',
     'wsgi.version': (1, 0)}

Anyone a hint? I tried DavDroid and Carddav clients that are supposed to work.

Richard

**Comment by pbiering** [comment]
Received same error via private e-mail from another radicale user.

At the end (after using strace) it turned out that there was a misconfiguration which isn't proper logged in the version you are using.

1. issue was regarding improper storage path, so check config and filesystem

```
[storage]
type = multifilesystem
filesystem_folder = /var/lib/radicale/collections
```

2. issue was regarding improper use use of radicale's filesystem layout in client configuration

here an example how to create calender/event folders manually for radicale >= 2

```
mkdir -p /var/lib/radicale/collections/collection-root/USER/mycalendar
echo '{"tag": "VCALENDAR"}' > /var/lib/radicale/collections/collection-root/USER/mycalendar/.Radicale.props

mkdir -p /var/lib/radicale/collections/collection-root/USER/myaddressbook
echo '{"tag": "VADDRESSBOOK"}'  > /var/lib/radicale/collections/collection-root/USER/myaddressbook/.Radicale.props
```

 @liZe: if there is some missing error detection code, it should be added to avoid strange messages. In addition, please add manual folder creation hints to documentation.

**Comment by rlucassen2** [comment]
Ok, thnx! I try to make it work and indeed, Radicale complains less, but apparently the version 2.0.rc0 needs a different documentation, right or wrong? I created these dirs:

    /var/lib/radicale/collections/collection-root/richard/addressbook/
    /var/lib/radicale/collections/collection-root/richard/calendar/

and I also created the appropiate .Radical.props in these directories. But when syncing, I only get the first entry from the addressbook + I get the same error on 1 search.

The vcf file is under the collection-root structure::

    /var/lib/radicale/collections/collection-root/richard/addressbook/cellphone.vcf

 I run radicale under "runit" and apparently it logs to two different log files (I'll have to take a look at that), but I concatenated them [here](http://tmp.xaq.nl/log.txt)

Any thoughts?

R.

**Comment by rlucassen2** [comment]
Seems that Radicale does not want to serve more than one single entry. I tried CardDAV and DavDroid. Is this a setting issue?

R.

**Comment by liZe** [comment]
> BTW: in the config file there is an outcommented option

Yes, it's been fixed in #553 (not merged yet). The only supported value is `multifilesystem`.

> Ok, thnx! I try to make it work and indeed, Radicale complains less, but apparently the version 2.0.rc0 needs a different documentation, right or wrong?

Right, you can find more information in #372 and in the [migration guide](http://kozea.github.io/Radicale/1to2/).

**Comment by rlucassen2** [comment]
Ok, thnx, I missed the migration guide. That was the missing link :-)

R.
